tests/store/test_artifact_repository_registry.py

```diff
@@ -0,0 +1,153 @@
+import mock
+import pytest
+from six.moves import reload_module as reload
+
+import mlflow
+from mlflow.store.artifact_repository_registry import ArtifactRepositoryRegistry
+from mlflow.store.rest_store import RestStore
+from mlflow.store.sqlalchemy_store import SqlAlchemyStore
+
+
+def test_standard_artifact_registry():
+    mock_entrypoint = mock.Mock()
+    mock_entrypoint.name = "mock-scheme"
+
+    with mock.patch(
+        "entrypoints.get_group_all", return_value=[mock_entrypoint]
+    ):
+        # Entrypoints are registered at import time, so we need to reload the
+        # module to register the entrypoint given by the mocked
+        # extrypoints.get_group_all
+        reload(mlflow.store.artifact_repository_registry)
+
+        expected_artifact_repository_registry = {
+            '',
+            's3',
+            'gs',
+            'wasbs',
+            'ftp',
+            'sftp',
+            'dbfs',
+            'mock-scheme'
+        }
+
+    assert expected_artifact_repository_registry.issubset(
+        mlflow.store.artifact_repository_registry._artifact_repository_registry._registry.keys()
+    )
+
+
+@pytest.mark.large
+def test_plugin_registration_via_installed_package():
+    """This test requires the package in tests/resources/mlflow-test-plugin to be installed"""
+
+    reload(mlflow.store.artifact_repository_registry)
+
+    assert (
+        "file-plugin" in
+        mlflow.store.artifact_repository_registry._artifact_repository_registry._registry
+    )
+
+    from mlflow_test_plugin import PluginLocalArtifactRepository
+
+    test_uri = "file-plugin:test-path"
+
+    plugin_repo = mlflow.store.artifact_repository_registry.get_artifact_repository(test_uri)
+
+    assert isinstance(plugin_repo, PluginLocalArtifactRepository)
+    assert plugin_repo.artifact_uri == test_uri
```

Comment:
Similar to what we did for the tracking store, it might be nice to check a property we explicitly set (e.g., `is_plugin = True`) as opposed to making an assumption about the field variables set in LocalArtifactRepository.

Reply:
Good catch, done.